planner/core/task.go

```diff
@@ -1003,7 +1003,7 @@ func (p *PhysicalTopN) getPushedDownTopN(childPlan PhysicalPlan) *PhysicalTopN {
 	topN := PhysicalTopN{
 		ByItems: newByItems,
 		Count:   newCount,
-	}.Init(p.ctx, stats, p.blockOffset)
+	}.Init(p.ctx, stats, p.blockOffset, p.GetChildReqProps(0))
```

Comment:
Why do we change it?

Reply:
it is related to https://github.com/pingcap/tidb/pull/21459#discussion_r549559676, because the prop is accessed later, but it is nil.